packages/fast-components-react-msft/src/call-to-action/call-to-action.spec.tsx

```diff
@@ -0,0 +1,116 @@
+        const props: ICallToActionHandledProps = {
+            appearance: CallToActionAppearance.primary
+        };
+
+        const rendered: any = shallow(
+            <Component {...props}/>
+        );
+
+        const callToAction: any = rendered.first().shallow();
+        // Get the expected className value from the list of generated managed classes
+        const expectedClassName: string = callToAction.instance().props.managedClasses.callToAction_backer;
+        const rootClassName: string = callToAction.instance().props.managedClasses.callToAction;
+
+        expect(callToAction.instance().props.appearance).toEqual(CallToActionAppearance.primary);
+        // Generated managedClass should be passed to className
+        expect(callToAction.prop("className")).toBe(`${rootClassName} ${expectedClassName}`);
+    });
+
+    // tslint:disable-next-line:max-line-length
+    test("should set a className that matches callToAction_lightweight managedClass when ButtonAppearance.lightweight is passed to the appearance prop", () => {
+        const props: ICallToActionHandledProps = {
+            appearance: CallToActionAppearance.lightweight
+        };
+
+        const rendered: any = shallow(
+            <Component {...props}/>
+        );
+
+        const callToAction: any = rendered.first().shallow();
+        // Get the expected className value from the list of generated managed classes
+        const expectedClassName: string = callToAction.instance().props.managedClasses.callToAction_lightweight;
+        const rootClassName: string = callToAction.instance().props.managedClasses.callToAction;
+
+        expect(callToAction.instance().props.appearance).toEqual(CallToActionAppearance.lightweight);
+        // Generated managedClass should be passed to className
+        expect(callToAction.prop("className")).toBe(`${rootClassName} ${expectedClassName}`);
+    });
+
+    // tslint:disable-next-line:max-line-length
+    test("should set a className that matches callToAction_lightweight managedClass when ButtonAppearance.justified is passed to the appearance prop", () => {
+        const props: ICallToActionHandledProps = {
+            appearance: CallToActionAppearance.justified
+        };
+
+        const rendered: any = shallow(
+            <Component {...props}/>
+        );
+
+        const callToAction: any = rendered.first().shallow();
+        // Get the expected className value from the list of generated managed classes
+        const expectedClassName: string = callToAction.instance().props.managedClasses.callToAction_lightweight;
+        const rootClassName: string = callToAction.instance().props.managedClasses.callToAction;
+
+        expect(callToAction.instance().props.appearance).toEqual(CallToActionAppearance.justified);
+        // Generated managedClass should be passed to className
+        expect(callToAction.prop("className")).toBe(`${rootClassName} ${expectedClassName}`);
+    });
```

Comment:
Can we add two tests? One to ensure that a custom className can be passed when the appearance prop is set, and one that ensures that it can be passed when it is NOT set?

Reply:
Sure

Reply:
@chrisdholt tests added